server/controllers/eventsController.js

```diff
@@ -10,7 +10,133 @@ const agent = new https.Agent({
   rejectUnauthorized: false,
 });
 
+const getRandom = (x) => {
+  return Math.floor(Math.random() * x);
+};
+// get min threhold from discount list
+const getMinThreshold = (discountList) => {
+  let minThreshold = discountList[0].threshold;
+  discountList.forEach((discount) => {
+    if (discount.threshold < minThreshold) minThreshold = discount.threshold;
+  });
+  return minThreshold;
+};
+
 const eventsController = {
+  startDraw: async (req, res, next) => {
+    const { member_id, event_id } = req.body;
+    const TOKEN = req.headers.authorization;
+    validator.existValidate(member_id, "member_id", next);
+    validator.existValidate(event_id, "event_id", next);
+    axios
+      .get(
+        `http://member-api.appworks.local/api/v1/admin/members/${member_id}`,
+        { httpsAgent: agent }
+      )
+      .then(async (response) => {
+        // 檢查是否加入三個月
+        const register_at = new Date(response.data.data.register_at);
+        const now = new Date();
+        if (now.getTime() - register_at.getTime() > 1000 * 60 * 60 * 24 * 90)
+          throw next(appError(200, `使用者註冊未滿三個月`, "102", next));
+        else {
+          // 拿取會員近一個月消費金額
+          axios
+            .get(`http://order-api.appworks.local/api/v1/admin/orders`, {
+              httpsAgent: agent,
+              params: { member_id: member_id },
+            })
+            .then(async (response) => {
+              // timestamp of one month age
+              const oneMonthAgo =
+                new Date().getTime() - 30 * 24 * 60 * 60 * 1000;
```

Comment:
Nit: You can use dayjs to format the date and time, which will make it more understandable.